anchor-cli.patch

```diff
@@ -1,45 +0,0 @@
-diff --git a/idl/src/build.rs b/idl/src/build.rs
-index 96dc3db2..1b098718 100644
---- a/idl/src/build.rs
-+++ b/idl/src/build.rs
-@@ -60,16 +60,8 @@ pub fn build_idl(
-
- /// Build IDL.
- fn build(program_path: &Path, resolution: bool, skip_lint: bool, no_docs: bool) -> Result<Idl> {
--    // `nightly` toolchain is currently required for building the IDL.
--    let toolchain = std::env::var("RUSTUP_TOOLCHAIN")
--        .map(|toolchain| format!("+{}", toolchain))
--        .unwrap_or_else(|_| "+nightly".to_string());
--
--    install_toolchain_if_needed(&toolchain)?;
--
-     let output = Command::new("cargo")
-         .args([
--            &toolchain,
-             "test",
-             "__anchor_private_print_idl",
-             "--features",
-@@ -202,23 +194,6 @@ fn build(program_path: &Path, resolution: bool, skip_lint: bool, no_docs: bool)
-     idl.ok_or_else(|| anyhow!("IDL doesn't exist"))
- }
-
--/// Install the given toolchain if it's not already installed.
--fn install_toolchain_if_needed(toolchain: &str) -> Result<()> {
--    let is_installed = Command::new("cargo")
--        .arg(toolchain)
--        .output()?
--        .status
--        .success();
--    if !is_installed {
--        Command::new("rustup")
--            .args(["toolchain", "install", toolchain.trim_start_matches('+')])
--            .spawn()?
--            .wait()?;
--    }
--
--    Ok(())
```

Comment:
wouldn't anchor try to install the toolchain again and fail with disk access without this patch?

Reply:
The code has changed, so it's no longer calling into rustup.

However, it's doing something that's just as problematic, by force-installing a different version of the solana / anchor tools if a different version is detected: https://github.com/coral-xyz/anchor/blob/f86a9c97fdd39a9c8675a6c606a26148bcbf6297/cli/src/lib.rs#L554

We can have a patch to either remove that entirely, or change it to return an error if a version mismatch is detected. Happy to go either way! It might be nice to even upstream that change

Reply:
yeah lets just remove it to return an error, last time Anchor was doing something like this, the failure would just print a cryptic disk access error so it was very hard to understand the issue. I wish these tools would stop trying to auto fetch dependencies and ask the user to install the specific requirement

Reply:
> I wish these tools would stop trying to auto fetch dependencies and ask the user to install the specific requirement

I think you can imagine the amount of bugs and user questions we've had to answer because of this kind of behavior :upside_down_face: 

> If we're certain this error won't trigger, maybe a comment in the derivation so future updaters would know this is a potential pitfall would suffice

The error can definitely trigger if a program has a different version of Anchor / Solana declared, so best to be clear. I added a patch to return that error.